tests/Microsoft.Identity.Test.Unit/PublicApiTests/ExtraBodyParametersTests.cs

```diff
@@ -0,0 +1,295 @@
+                        .WithFmiPath("SomeFmiPath/FmiCredentialPath") //Sets fmi path in client credential request.
+                        .WithExtraBodyParameters(extraBodyParams) //Sets attributes in client credential request.
+                        .ExecuteAsync()
+                        .ConfigureAwait(false);
+
+                //Assert
+                Assert.IsNotNull(authResult);
+                Assert.AreEqual(authResult.AuthenticationResultMetadata.TokenSource, TokenSource.IdentityProvider);
+            }
+        }
+
+        [TestMethod]
+        public async Task WithExtraBodyParameters_NullInput_ReturnToken()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                // Arrange
+                httpManager.AddInstanceDiscoveryMockHandler();
+                httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+                var confidentialApp = ConfidentialClientApplicationBuilder
+                            .Create(_clientId)
+                            .WithAuthority("https://login.microsoftonline.com/", _tenantId)
+                            .WithClientSecret("ClientSecret")
+                            .WithHttpManager(httpManager)
+                            .WithExperimentalFeatures(true) // Enable experimental features to use WithExtraBodyParameters
+                            .BuildConcrete();
+
+                // Act & Assert
+                // Ensure that the token is returned even when no extra body parameters are provided
+                var authResult = await confidentialApp.AcquireTokenForClient(new[] { _scope })
+                    .WithExtraBodyParameters(null)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                Assert.IsNotNull(authResult);
+                Assert.AreEqual(authResult.AuthenticationResultMetadata.TokenSource, TokenSource.IdentityProvider);
+
+                authResult = await confidentialApp.AcquireTokenForClient(new[] { _scope })
+                    .WithExtraBodyParameters(null)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                Assert.IsNotNull(authResult);
+                Assert.AreEqual(authResult.AuthenticationResultMetadata.TokenSource, TokenSource.Cache);
+
+                //Ensure that the token can still be retrieved from the cache when the input is an empty dictionary
+                authResult = await confidentialApp.AcquireTokenForClient(new[] { _scope })
+                    .WithExtraBodyParameters(new Dictionary<string, Func<CancellationToken, Task<string>>>())
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                Assert.IsNotNull(authResult);
+                Assert.AreEqual(authResult.AuthenticationResultMetadata.TokenSource, TokenSource.Cache);
+            }
+        }
+
+        private Task<string> GetPdpAuthorization()
```

Comment:
What does Pdp mean?

Reply:
Ah, this is part of the attribute tokens flow in MISE, No reason to have it here, I can rename this method

Reply:
Sure, I was a bit confused with the term. Thanks